File: backend/internal/models/recommendation.go
package models

type Recommendation struct {
	ID         int    `json:"id"`
	FarmID     int    `json:"farm_id"`
	Crop       string `json:"crop"`
	Type       string `json:"type"`
	Message    string `json:"message"`
	Severity   string `json:"severity"`
	Priority   string `json:"priority"`
	Confidence string `json:"confidence"`
	Reason     string `json:"reason"`
	CreatedAt  string `json:"created_at"`
}

File: frontend/lib/models/recommendation.dart
// lib/models/recommendation.dart — Recommendation data model.
//
// Fields:
//   int farmId, String type, String message, String severity, String createdAt
//
// Responsibilities:
//   - Define a Recommendation class with a fromJson() factory constructor.
//   - JSON keys match the Go backend GET /api/recommendation response.
//
// Used by:
//   services/recommendation_service.dart — deserializes API response.
//   screens/dashboard/, screens/farm_details/ — shows recommendations.
//   widgets/recommendation_card.dart — renders each recommendation.
class Recommendation {}